nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/framework/configuration/FlowControllerConfiguration.java

```diff
@@ -471,4 +479,38 @@ public AssetSynchronizer assetSynchronizer() throws Exception {
     public AssetComponentManager affectedComponentManager() throws Exception {
         return new StandardAssetComponentManager(flowController());
     }
+
+    /**
+     * Flow Action Reporter configured from NiFi Application Properties
+     *
+     * @return Flow Action Reporter
+     * @throws Exception Thrown on failures to create Flow Action Reporter
+     */
+    @Bean(destroyMethod = "preDestruction")
+    public FlowActionReporter flowActionReporter() throws Exception {
+        final String configuredClassName = properties.getProperty(FLOW_ACTION_REPORTER_IMPLEMENTATION, DEFAULT_FLOW_ACTION_REPORTER_IMPLEMENTATION);
+        if (StringUtils.isBlank(configuredClassName)) {
+            throw new RuntimeException("Cannot create FlowActionReporter because the configuration is missing the following property: "
```

Comment:
Recommend a more specific exception:
```suggestion
            throw new IllegalStateException("Cannot create FlowActionReporter because the configuration is missing the following property: "
```